opencog/nlp/relex2logic/det.scm

```diff
@@ -0,0 +1,59 @@
+
+(define det
+    (BindLink
+        (VariableList
+            (TypedVariableLink
+                (VariableNode "$a-parse")
+                (TypeNode "ParseNode")
+            )
+            (TypedVariableLink
+                (VariableNode "$noun")
+                (TypeNode "WordInstanceNode")
+            )
+            (TypedVariableLink
+                (VariableNode "$det")
+                (TypeNode "WordInstanceNode")
+            )
+        )
+        (AndLink
+            (WordInstanceLink
+                (VariableNode "$noun")
+                (VariableNode "$a-parse")
+            )
+            (WordInstanceLink
+                (VariableNode "$det")
+                (VariableNode "$a-parse")
+            )
+            (EvaluationLink
+                (DefinedLinguisticRelationshipNode "_det")
+                (ListLink
+                    (VariableNode "$noun")
+                    (VariableNode "$det")
+                )
+            )
+            (InheritanceLink 
+                (VariableNode "$noun")
```

Comment:
@williampma we agreed on iterating over all rules and apply them on the focus set in one call or step. This reminds me of the policy vs mechanism isolation principle that @linas mentioned on another thread. As @AmeBel  said this particular case can be handled by having N rule bases for the N phase and call FC N times which seems like a policy to my understanding.